CDCgov/prime-reportstream
# 5746 Generate HL7 SFT segment from FHIR

This PR generates the SFT segment from FHIR data.  It also fixes some bugs found in the process of generating the config.

Test Steps:
1. Check out the branch for this PR.
2. Verify the tests pass by running the command `./gradlew package` and verifying the build is successful.
3. Run the following command to convert a FHIR bundle to HL7.  
`./prime fhirdata --input-file "src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.fhir" -s metadata/hl7_mapping/ORU_R01/ORU_R01-base.yml`
4. Verify an HL7 message is generated from the conversion and includes the SFT segment.
5. Look at the source HL7 used at `src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.hl7` and verify the data for the SFT segment is identical with the following exceptions:
  a. SFT-6 requires #6104, but once that issue is merged the date will appear

## Changes
- Added initial set of configs for the FHIR to HL7 conversion
- Had to remove the FHIR path validation at the schema level as some paths will use custom constants that are not available at that time and hence the validation will fail.  Validation will now happen at conversion time instead of schema load time.
- Fixed bug where the top level schema constants were not loading and made it so the arguments of functions will require a custom context to eliminate the possibility of a coding mistake
- Fixed get value not stopping at the first valid value
- Added try/catch around evaluates.  I was getting an index out of bound exception when trying to fetch an extension that did not exist.
- Added support for prefix constants.  This is the same feature that the FHIR Path Engine has for the ext constant.  If you use a constant in FHIR path it can replace a prefix.  E.g., `%'rsext-suffix'` will be converted to `<rsextvalue>suffix`.  
- Fixed an issue where the constant resolver is called at all times and incorrect error out.  Now it returns null instead of an exception

## Checklist

### Testing

### Process

## Linked Issues
- #5746

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
